utils/oominfo/oomparser.go

```diff
@@ -0,0 +1,200 @@
+}
+
+/* determines if a line is a specific pattern. If so it will contain the PID,
+process name, and time of death of the OOM kill instance.  Called by 
+readMessages */
+func checkIfKillConfirmation(line string) (bool, error) {
+	/* todo: glean container name(s), possibly in new function and not from the
+	same line */
+	matched, err := regexp.MatchString(".*] Killed process .*", line)
+	if matched {
+		return true, err
+	}
+	return false, err
+
+}
+
+// checks if a line contains words relating to oom kills. called by readMessages
+func checkForMentionOfOOM(currentLine string) (bool) {
+	//todo: use regexp
+	hitlist := []string{ "oom", "kill", "Kill", "out of memory" }
+	words := strings.Fields(currentLine)
+	for i := 0; i < len(words); i++ {
+		word := words[i]
+		for j := 0; j < len(hitlist); j++ {
+			hit := hitlist[j]
+			if strings.Contains(word, hit) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// records log messages related to OOM kills in a file. called by readMessages
+func writeLineToOOMFileLog(line string, path string) error {
+	// open file or initialize it if it doesn't exist
+	file, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
+		if err != nil {
+		return err
+	}
+	defer file.Close()
+	w := bufio.NewWriter(file)
+	_, err = w.WriteString(line)
+	if err != nil {
+		log.Println("oomparser/writeLine could not write OOM to file: %v", err)
+	}
+	return w.Flush()
+}
+
+/* opens a reader to grab new messages from the exec.Command.StdoutPipe object
+called outPipe passed in from main().  It reads line by line splitting on 
+the "\n" character.  Every line is checked for relevance to the oomLogFile
+If it contains certain words such as "oom" or "kill" it is added. Then the
+lines are checked against a regexp to determine if the pid, process name, etc
+can be gleaned. If so readMessages then calls a function that makes an 
+oomInstance object from the line and adds it to the oomLog */
+func readMessages(outPipe io.ReadCloser, oomLogFile string, oomLog *oomLogList) error {
+	ioreader := bufio.NewReader(outPipe)
+	line, err := ioreader.ReadString('\n')
+
+	for err == nil {
+		is_oom_message := checkForMentionOfOOM(line)
+		if is_oom_message {
+			err = writeLineToOOMFileLog(line, oomLogFile)
+			if err != nil {
+				log.Println("oomparser/readMessages failed to writeLineToOOMFileLog:\n", err)
+			}
+		}
+		is_oom_message, err = checkIfKillConfirmation(line)
+		if is_oom_message {
+			err = oomLog.addToOOMLog(line)
+			if err != nil {
+				log.Println("oomparser/readMessages failed to addToOOMLog:\n", err)
+			} else {
+				log.Println("Added an oomInstance to oomLog")
+				oomLog.PrettyPrintOOMLog()
+			}
+		}
+		if err != nil {
+			log.Println("oomparser/readMessages failed at checkIfKillConfirmation:\n", err)
+
+		}
+		line, err = ioreader.ReadString('\n')
+	}
+	return err
+}
+
+/* calls the "tail -f" command on "/var/log/messages" to search for logs that
+pertain to OOM kills */
+func main() {
```

Comment:
Lets get the main in a different file (with package main) and this library in this filw (with package oomparser). That way we can use the library and compile a tester.